File: Data Structures & Algorithms/longest-increasing-subsequence/submission-1.cpp
class Solution {
public:
    int lengthOfLIS(vector<int>& nums) {
        int n=nums.size();
        vector<int> dp(n,1);
        for(int i=0 ;i<n ;i++){
            int j=i-1;
            while(j>-1){
                if(nums[j]<nums[i])
                    dp[i]=max(dp[i],dp[j]+1);
                j--;
            }
            cout<<dp[i]<<" ";
        }
        return *max_element(dp.begin(),dp.end());
        
    }
};

File: Data Structures & Algorithms/task-scheduling/submission-1.py
class Solution:
    def leastInterval(self, tasks: List[str], n: int) -> int:
        freq = Counter(tasks)
        max_freq = max(freq.values())
        max_count = sum(1 for count in freq.values() if count == max_freq)
        ans = (max_freq - 1) * (n + 1) + max_count
        return max(len(tasks), ans)
        